src/main/java/htsjdk/variant/vcf/VCFEncoder.java

```diff
@@ -64,15 +66,32 @@ public void setAllowMissingFieldsInHeader(final boolean allow) {
 		this.allowMissingFieldsInHeader = allow;
 	}
 
+	/** encodes a new Variant context as VCF 
+	 * @return the VCF line
+	 */
 	public String encode(final VariantContext context) {
-		if (this.header == null) {
-			throw new NullPointerException("The header field must be set on the VCFEncoder before encoding records.");
-		}
+    	try {
+            return encode(context,new StringBuilder(1000)).toString();
+        } catch (final IOException error) {
+           throw new RuntimeIOException("Cannot encode variant", error);
+        }
+	}
 
-		final StringBuilder stringBuilder = new StringBuilder();
 
+   /** 
+    * encodes a new Variant context as VCF, writes it to a java.lang.Appendable 
+    * 
+    * @param context the variant
+    * @param vcfoutput the java.lang.Appendable
+    * @return the java.lang.Appendable 'vcfoutput'
+    * @throws IOException
+    */
+   public Appendable encode(final VariantContext context,final Appendable vcfoutput) throws IOException {	
```

Comment:
Can you change the signature to be the other way around? Will be useful for #822